packages/ragbits-document-search/src/ragbits/document_search/retrieval/rerankers/litellm.py

```diff
@@ -0,0 +1,63 @@
+from typing import List
+
+import litellm
+from pydantic import BaseModel
+
+from ragbits.document_search.documents.element import Element, TextElement
+from ragbits.document_search.retrieval.rerankers.base import Reranker
+
+
+class LiteLLMReranker(BaseModel, Reranker):
+    """
+    A LiteLLM reranker for providers such as Cohere, Together AI, Azure AI.
+    For more details and parameters see https://docs.litellm.ai/docs/rerank
+
+    Args:
+        model: The model to use.
+        top_n: The number of most relevant documents or indices to return.
+        return_documents: Whether to return the documents or just indices and relevalnce scores.
+        rank_fields: The fields to rank on.
+        max_chunks_per_doc: The maximum number of chunks per document to consider.
+
+    Returns:
+        The reranked chunks.
+    """
+
+    model: str
+    top_n: int | None = None
```

Comment:
These params should be in a separate data class (look at the `LLMOptions` implementation)